Formula/purescript.rb

```diff
@@ -18,13 +18,25 @@ class Purescript < Formula
   depends_on "ghc" => :build
   depends_on "cabal-install" => :build
 
-  def install
-    # Fix "error: Couldn't match type 'Text' with 'Line'"
-    # Upstream issue "turtle 1.3 breaks build"
-    # Reported 10 Dec 2016 https://github.com/purescript/purescript/issues/2472
-    inreplace "purescript.cabal", "turtle -any", "turtle < 1.3"
+  # Fix "Couldn't match type '[Char]' with 'Text'"
+  # Upstream issue from 2 Jan 2017 https://github.com/purescript/purescript/issues/2528
+  resource "purescript-cabal-hackage" do
+    url "https://hackage.haskell.org/package/purescript-0.10.4/purescript.cabal"
+    sha256 "a5dacd7a8e23b2aaa2e0f606372496d44cdb9217dbb565b06ce584a22f986a16"
+  end
 
-    install_cabal_package :using => ["alex", "happy"]
+  def install
+    # overwrites purescript.cabal
+    buildpath.install resource("purescript-cabal-hackage")
+
+    cabal_sandbox do
+      system "cabal", "get", "turtle-1.2.8"
+      inreplace "turtle-1.2.8/turtle.cabal",
```

Comment:
Yep. That will work too. I usually do this because I want it to fail when the latest version changes but since we have to lock in turtle-1.2.8 anyway, that won't happen here anyway.